arelle/Locale.py

```diff
@@ -65,20 +70,23 @@ def getUserLocale(localeCode=''):
         conv = locale.localeconv() # use 'C' environment, e.g., en_US
     if C_LOCALE is None: # load culture-invariant C locale
         C_LOCALE = locale.localeconv()
-    return conv
+    return cast(LocaleDict, conv)
 
-def getLanguageCode():
+def getLanguageCode() -> str:
     if sys.platform == "darwin":
         # possibly this MacOS bug: http://bugs.python.org/issue18378
         # even when fixed, macOS won't provide right default code for some language-culture combinations
         localeQueryResult = subprocess.getstatusoutput("defaults read -g AppleLocale")  # MacOS only
         if localeQueryResult[0] == 0 and localeQueryResult[1]: # successful
             return localeQueryResult[1][:5].replace("_","-")
     import locale
-    try:
-        return locale.getdefaultlocale()[0].replace("_","-")
-    except (AttributeError, ValueError): #language code and encoding may be None if their values cannot be determined.
-        return "en"
+
+    language_code, encoding = locale.getdefaultlocale()
```

Comment:
It isn't conventional for python, but arelle uses camel case everywhere, we should stay consistent:
```suggestion
    languageCode, encoding = locale.getdefaultlocale()
```